backend/src/controllers/deck.js

```diff
@@ -0,0 +1,172 @@
+import deckService from '../services/deck.js';
+import userService from '../services/user.js';
+import cardService from '../services/card.js';
+
+import { Card } from '../db/index.js';
+
+export default {
+  /**
+   * Expresse.js controller for GET /decks
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   */
+  cget: async (req, res, next) => {
+    try {
+      const decks = await deckService.findAll();
+      res.json(decks);
+    } catch (err) {
+      next(err);
+    }
+  },
+  /**
+   * Express.js controller for POST /deck
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   */
+  post: async (req, res, next) => {
+    try {
+      const deck = await deckService.create({
+        name: req.body.name,
+        userId: req.user.id,
+      });
+      const createdDeck = await deckService.findById(deck.id);
+      res.status(201).json(createdDeck);
+    } catch (err) {
+      next(err);
+    }
+  },
+  /**
+   * Express.js controller for GET /decks/:id
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   * */
+  get: async (req, res, next) => {
+    try {
+      const deck = await deckService.findById(parseInt(req.params.id), {
+        include: Card,
+      });
+      if (!deck) return res.sendStatus(404);
```

Comment:
```suggestion
      if (!deck) throw new Error('Deck not found', { cause: 'Not Found' });
```

Ca va automatiquement renvoyer une 404

https://github.com/ESGI-69/challenge-node-vue/blob/51392b282c6e93d101e2305b15aa72ec5e1b5ea7/backend/src/errorHandler.js#L56-L57